### PR TITLE
Fix #1

### LINKED ISSUE #1: Security issue
Hi,

This config appears to leave ports exposed on the host. This is a security issue. Will soon submit PR with fixes

Davide

### PR DESCRIPTION
**Pull request by ddavidebor**
Fixes security issue, moves docker-compose v1 to v2 and a few other improvements.
Please see the extended commit logs for more info.

Davide

### CODE REVIEW

**Comment by ddavidebor** [comment]
I haven't tested it from scratch again, but i'm currently running this in production.

**Comment by thurloat** [comment]
thanks for the fixes & updates.

The exposed ports didn't effect me because the environment I'm running in requires explicitly allowing ports to be opened on the external port.

you might want to remove your github secret from this PR and resubmit.

I'm a bit concerned that the `depends_on` for sentry_web is overly strict, as the web process will work fine without the worker/cron processes running.

File: env/sentry.sample [hunk 1]
```diff
@@ -7,3 +7,6 @@ SENTRY_EMAIL_PASSWORD=<SMTP PASSWORD>
 SENTRY_EMAIL_USER=SMTP_Injection
 SENTRY_EMAIL_PORT=587
 SENTRY_EMAIL_USE_TLS=True
+
+GITHUB_APP_ID=123abc123abc123abc12
```
Comment (ddavidebor) [inline env/sentry.sample:11]: @thurloat just 123abc repeated

**Comment by ddavidebor** [comment]
@thurloat i just wrote "123abc" as an example of how to configure the github integration, wich i guess is a common user case

**Comment by thurloat** [comment]
@ddavidebor haha, I didn't even look. I saw the addition of the app id and secret and averted my eyes like standing next to someone typing their password. 🤦‍♂️

**Comment by ddavidebor** [comment]
lol

**Pull request merged**